packages/cli/scripts/start.js

```diff
@@ -693,18 +689,19 @@ function handleRunTypeCheck(srcPath, inputFiles, tsconfig, config) {
 }
 
 async function getHandlerFilePath(appPath, srcPath, handler) {
-  const parts = handler.split(".");
+  const parts = path.basename(handler).split(".");
+  const dir = path.dirname(handler);
   const name = parts[0];
 
   const extensions = [".ts", ".tsx", ".jsx"];
   for (const ext of extensions) {
-    const file = path.join(appPath, srcPath, `${name}${ext}`);
+    const file = path.join(appPath, srcPath, dir, `${name}${ext}`);
     if (await checkFileExists(file)) {
       return file;
     }
   }
 
-  return path.join(appPath, srcPath, `${name}.js`);
+  return path.join(appPath, srcPath, dir, `${name}.js`);
```

Comment:
Curious why it defaults to this without verifying it exists

Reply:
Yeah we should. Let me add a check.